feat(template): add open archiver template

wayneshn opened the pull request:
## Changes

- Add Open Archiver template. 
- Open Archiver is a popular open-source [email-archiving solution](https://github.com/LogicLabs-OU/OpenArchiver).

wayneshn committed:
add openarchiver.svg

wayneshn committed:
Add files via upload

wayneshn committed:
Create open-archiver.yml

ShadowArcanist commented on templates/compose/open-archiver.yml:
```diff
@@ -0,0 +1,69 @@
+    open-archiver:
+        image: logiclabshq/open-archiver:latest
+        environment:
+            - SERVICE_FQDN_3000
```
Use `SERVICE_URL_3000` instead of `SERVICE_FQDN_3000` because on recent update a long time bug has been fixed (bug is the env `SERVICE_FQDN` gives values with the https protocol instead of just giving out the FQDN like example.com )

ShadowArcanist commented on templates/compose/open-archiver.yml:
```diff
@@ -0,0 +1,69 @@
+            - PORT_FRONTEND=${PORT_FRONTEND:-3000}
+            - NODE_ENV=${NODE_ENV:-production}
+            - SYNC_FREQUENCY=${SYNC_FREQUENCY:-* * * * *}
+            - POSTGRES_DB=${POSTGRES_DB:-open_archive}
+            - POSTGRES_USER=${POSTGRES_USER:-admin}
+            - POSTGRES_PASSWORD=${SERVICE_PASSWORD_POSTGRES}
```
Why these DB env are added here with fallback values? There are already defined on `postgres` service no?

weishest replied:
I will keep the fallback values here and remove the ones from the `postgres` service.

ShadowArcanist commented on templates/compose/open-archiver.yml:
```diff
@@ -0,0 +1,69 @@
+            - RATE_LIMIT_WINDOW_MS=${RATE_LIMIT_WINDOW_MS:-60000}
+            - RATE_LIMIT_MAX_REQUESTS=${RATE_LIMIT_MAX_REQUESTS:-100}
+        volumes:
+            - ${COOLIFY_VOLUME_APP_DATA}:/var/data/open-archiver
```
What is this `${COOLIFY_VOLUME_APP_DATA}` do? This env is not used anywhere else on this compose

ShadowArcanist commented on templates/compose/open-archiver.yml:
```diff
@@ -0,0 +1,69 @@
+            - POSTGRES_PASSWORD=${SERVICE_PASSWORD_POSTGRES}
+            - LC_ALL=C
+        volumes:
+            - ${COOLIFY_VOLUME_POSTGRES}:/var/lib/postgresql/data
```
What is this `${COOLIFY_VOLUME_POSTGRES}` do? This env is not used anywhere else on this compose

ShadowArcanist commented on templates/compose/open-archiver.yml:
```diff
@@ -0,0 +1,69 @@
+        image: valkey/valkey:8-alpine
+        command: valkey-server --requirepass ${SERVICE_PASSWORD_VALKEY}
+        volumes:
+            - ${COOLIFY_VOLUME_VALKEY}:/data
```
What is this `${COOLIFY_VOLUME_VALKEY}` do? This env is not used anywhere else on this compose

ShadowArcanist commented on templates/compose/open-archiver.yml:
```diff
@@ -0,0 +1,69 @@
+        environment:
+            - MEILI_MASTER_KEY=${SERVICE_PASSWORD_MEILISEARCH}
+        volumes:
+            - ${COOLIFY_VOLUME_MEILISEARCH}:/meili_data
```
What is this `${COOLIFY_VOLUME_MEILISEARCH}` do? This env is not used anywhere else on this compose

ShadowArcanist commented on templates/compose/open-archiver.yml:
```diff
@@ -0,0 +1,69 @@
+            - valkey
+            - meilisearch
+
+    postgres:
+        image: postgres:17-alpine
```
Adding healthcheck for db would be helpful since other services on this compose depends on this postgres service

ShadowArcanist commented:
Also you have to send PR to `next` branch, not `v4.x`

wayneshn reviewed (commented):
Resolved comments. submitting a new version

wayneshn committed:
Update open-archiver.yml

- Removed extra fallback values from DB environment variables
- Removed unnecessary environment variables for data volume
- Adding healthcheck for PG service

wayneshn commented:
@ShadowArcanist the code has been updated based on your feedback. Would you please take another look?

ShadowArcanist reviewed (approved):
Looks good to me!. 

Let's wait for one of the core maintainers (Andras or Peak) to do a final review and merge the PR. 

I don’t have an ETA for their response, so we’ll need to be patient and avoid pinging them for attention.

wayneshn committed:
Merge branch 'next' into v4.x

github-actions[bot] committed:
docs: update changelog

ShadowArcanist commented on templates/compose/open-archiver.yml:
```diff
@@ -0,0 +1,77 @@
+    open-archiver:
+        image: logiclabshq/open-archiver:latest
+        environment:
+            - SERVICE_URL_3000
```
```suggestion
            - SERVICE_URL_OPEN_ARCHIVER_3000
```

ShadowArcanist commented on templates/compose/open-archiver.yml:
```diff
@@ -0,0 +1,77 @@
+        image: logiclabshq/open-archiver:latest
+        environment:
+            - SERVICE_URL_3000
+            - SERVICE_URL=${SERVICE_URL_3000}
```
```suggestion
```

ShadowArcanist commented on templates/compose/open-archiver.yml:
```diff
@@ -0,0 +1,77 @@
+            - NODE_ENV=${NODE_ENV:-production}
+            - SYNC_FREQUENCY=${SYNC_FREQUENCY:-* * * * *}
+            - POSTGRES_DB=${POSTGRES_DB:-open_archive}
+            - POSTGRES_USER=${POSTGRES_USER:-admin}
```
```suggestion
            - 'POSTGRES_USER=${SERVICE_USER_POSTGRES}'
```

ShadowArcanist commented on templates/compose/open-archiver.yml:
```diff
@@ -0,0 +1,77 @@
+            - POSTGRES_DB=${POSTGRES_DB:-open_archive}
+            - POSTGRES_USER=${POSTGRES_USER:-admin}
+            - POSTGRES_PASSWORD=${SERVICE_PASSWORD_POSTGRES}
+            - DATABASE_URL=postgresql://${POSTGRES_USER}:${POSTGRES_PASSWORD}@postgres:5432/${POSTGRES_DB}
```
```suggestion
            - 'DATABASE_URL=postgresql://${SERVICE_USER_POSTGRES}:${SERVICE_PASSWORD_POSTGRES}@postgres:5432/${POSTGRES_DB}'
```

ShadowArcanist commented on templates/compose/open-archiver.yml:
```diff
@@ -0,0 +1,77 @@
+        image: postgres:17-alpine
+        environment:
+            - POSTGRES_DB=${POSTGRES_DB}
+            - POSTGRES_USER=${POSTGRES_USER}
```
```suggestion
            - 'POSTGRES_USER=${SERVICE_USER_POSTGRES}'
```

ShadowArcanist commented on templates/compose/open-archiver.yml:
```diff
@@ -0,0 +1,77 @@
+    postgres:
+        image: postgres:17-alpine
+        environment:
+            - POSTGRES_DB=${POSTGRES_DB}
```
```suggestion
            - 'POSTGRES_DB=${POSTGRES_DB:-open_archive}'
```

ShadowArcanist commented on templates/compose/open-archiver.yml:
```diff
@@ -0,0 +1,77 @@
+
+services:
+    open-archiver:
+        image: logiclabshq/open-archiver:latest
```
Please pin to a static version rather than 'latest' to prevent the template from breaking due to upstream changes.

```suggestion
        image: logiclabshq/open-archiver:v0.3.4
```

ShadowArcanist commented on templates/compose/open-archiver.yml:
```diff
@@ -0,0 +1,77 @@
+# documentation: https://openarchiver.com
```
Point to the actual docs url
```suggestion
# documentation: https://docs.openarchiver.com/
```

ShadowArcanist reviewed (changes requested):
It took me an hour to figure out why the app is not working...

- Please implement the requested changes
- Please resolve the conflicts (mostly caused by the changelog file)
- Please send PR to docs repo (check github actions bot comment below for more info)

peaklabs-dev committed:
Merge branch 'next' into v4.x

github-actions[bot] committed:
docs: update changelog

peaklabs-dev committed:
feat(service): improve open-archiver

- add STORAGE_ENCRYPTION_KEY
- add healthcheck for meilisearch and valkey
- fix ENCRYPTION_KEY format (needs to be HEX)
- fix open-archiver URL
- fix docs link
- fix formatting, order and naming
- fix indentation
- use yaml instead of yml

peaklabs-dev committed:
Merge branch 'v4.x' of https://github.com/wayneshn/coolify into pr/6593

github-actions[bot] committed:
docs: update changelog

peaklabs-dev reviewed (approved):
Thanks for the PR. I improved it quite a bit. Next time please do not open a PR from your forks `v4.x` or `next` branch
